### PR TITLE
pkgconf: init at 1.5.4

### PR DESCRIPTION
**Pull request by zaninime**
###### Motivation for this change
Add a new derivation.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/tools/misc/pkgconf/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchgit, automake, autoconf, libtool }:
+
+stdenv.mkDerivation rec {
+  name = "pkgconf-1.5.4";
+
+  src = fetchgit {
+    url = "https://git.dereferenced.org/pkgconf/pkgconf.git";
+    rev = "74133eda31bc1ed5947b4a3a854001e320b6c1fe";
```
Comment (jtojnar) [inline pkgs/development/tools/misc/pkgconf/default.nix]:
```suggestion
    rev = "refs/tags/${name}";
```

File: pkgs/development/tools/misc/pkgconf/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchgit, automake, autoconf, libtool }:
+
+stdenv.mkDerivation rec {
+  name = "pkgconf-1.5.4";
+
+  src = fetchgit {
+    url = "https://git.dereferenced.org/pkgconf/pkgconf.git";
```
Comment (jtojnar) [inline pkgs/development/tools/misc/pkgconf/default.nix]: No need to quote simple URLs.

File: pkgs/development/tools/misc/pkgconf/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  preConfigurePhases = ["autogenPhase"];
+
+  autogenPhase = ''
+    ./autogen.sh
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Package compiler and linker metadata toolkit";
+    homepage = https://git.dereferenced.org/pkgconf/pkgconf;
```
Comment (jtojnar) [inline pkgs/development/tools/misc/pkgconf/default.nix]: https://pkgconf.org/

File: pkgs/development/tools/misc/pkgconf/default.nix (hunk 2 above)
Comment (jtojnar) [reply pkgs/development/tools/misc/pkgconf/default.nix]: The website points to https://distfiles.dereferenced.org/pkgconf/pkgconf-1.1.0.tar.xz

File: pkgs/development/tools/misc/pkgconf/default.nix (hunk 3 above)
Comment (zaninime) [reply pkgs/development/tools/misc/pkgconf/default.nix]: The website is clearly out-of-date and returns an invalid HTTPS certificate. I would recommend against linking it in the derivation. It also points to GitHub, which is not the official repository any longer.

File: pkgs/development/tools/misc/pkgconf/default.nix (hunk 2 above)
Comment (jtojnar) [reply pkgs/development/tools/misc/pkgconf/default.nix]: See the bottom of the readme.

File: pkgs/development/tools/misc/pkgconf/default.nix (hunk 3 above)
Comment (zaninime) [reply pkgs/development/tools/misc/pkgconf/default.nix]: Any more concerns on this? The rest should be fixed.

**Comment by jtojnar** [comment]
Could you try reporting the broken website in their issue tracker?

**Comment by zaninime** [comment]
> Could you try reporting the broken website in their issue tracker?

https://git.dereferenced.org/pkgconf/pkgconf/issues/16

I will follow up on this if any news comes by. Should this PR be merged anyway, and I submit a new one when / if further information is available?

**Comment by jtojnar** [comment]
@GrahamcOfBorg build pkgconf

**Comment by jtojnar** [comment]
Thanks. we will fix the website when they respond.

**Pull request merged**